### PR TITLE
SAAS-418 modal library component

### PR DESCRIPTION
**Pull request by atymchuk**

### CODE REVIEW

File: packages/platform-core/src/components/Modal/Modal.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        }
+      };
+
+      document.addEventListener('keydown', escapeHandler);
+
+      return () => document.removeEventListener('keydown', escapeHandler);
+    }
+
+    return undefined;
+  });
```
[the remark this answers is not in the document]
Comment (atymchuk) [reply packages/platform-core/src/components/Modal/Modal.tsx:35]: Looks like a miss. Corrected. 
It actually does re-render in pmm :)

[the remark this answers is not in the document]
Comment (tiagomotasantos) [reply packages/platform-core/src/components/Modal/Modal.tsx]: Yap, only realised now :/

**Review by tiagomotasantos — APPROVED** [review]

File: packages/platform-core/src/components/Modal/Modal.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (atymchuk) [reply packages/platform-core/src/components/Modal/Modal.tsx]: lol, it already smells migration :)

File: packages/platform-core/src/components/Modal/Modal.test.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+describe('Modal window::', () => {
+  it('should render modal successfully', () => {
+    const onClose = jest.fn();
+    const root = shallow(<Modal onClose={onClose} isVisible title="test" />);
+
+    expect(root.find('[data-qa="modal-background"]').length).toEqual(1);
+    expect(root.find('[data-qa="modal-body"]').length).toEqual(1);
+    expect(root.find('[data-qa="modal-close-button"]').length).toEqual(1);
+    expect(root.find('[data-qa="modal-content"]').length).toEqual(1);
```
Comment (nicolalamacchia) [inline packages/platform-core/src/components/Modal/Modal.test.tsx:13]: you can use the `dataQa` helper here

File: packages/platform-core/src/components/Modal/Modal.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        }
+      };
+
+      document.addEventListener('keydown', escapeHandler);
+
+      return () => document.removeEventListener('keydown', escapeHandler);
+    }
+
+    return undefined;
+  }, [closeOnEscape]);
```
Comment (nicolalamacchia) [inline packages/platform-core/src/components/Modal/Modal.tsx:35]: Looks like eslint is complaining about `onClose` not being in the dependencies

File: packages/platform-ui/README.md [hunk 4]
```diff
@@ -27,18 +27,10 @@ Your app is ready to be deployed!
...
-You don’t have to ever use `eject`. The curated feature set is suitable for small and middle deployments, and you shouldn’t feel obligated to use this feature. However we understand that this tool wouldn’t be useful if you couldn’t customize it when you are ready for it.
-
-## Learn More
-
-You can learn more in the [Create React App documentation](https://facebook.github.io/create-react-app/docs/getting-started).
-
-To learn React, check out the [React documentation](https://reactjs.org/).
+```json
+"proxy": "https://platform-dev.percona.com"
+```
```
Comment (nicolalamacchia) [inline packages/platform-ui/README.md:36]: This might be ok for a public README, but I think that internally we should consider running SaaS locally

**Review by nicolalamacchia — CHANGES_REQUESTED** [review]

File: packages/platform-core/src/components/Modal/Modal.test.tsx (hunk 2 above)
Comment (atymchuk) [reply packages/platform-core/src/components/Modal/Modal.test.tsx:13]: it's a nit, I want to go fast

File: packages/platform-ui/README.md (hunk 4 above)
Comment (atymchuk) [reply packages/platform-ui/README.md:36]: We didn't have either in the readme.

File: packages/platform-core/src/components/Modal/Modal.tsx (hunk 3 above)
Comment (atymchuk) [reply packages/platform-core/src/components/Modal/Modal.tsx]: Damn, my eslint is broken. I should have seen it.
Fixed.

**Review by nicolalamacchia — APPROVED** [review]

**Pull request merged**